Use SQLAlchemy 1.x

spenczar opened the pull request:
Dagster is incompatible with SQLAlchemy 2, and we use dagster to run precovery jobs. Unfortunately, this means we need to revert the changes in #66.

akoumjian commented on precovery/ingest/index_observations.py:
```diff
@@ -109,7 +108,6 @@ def index(
         completed += 1
         logger.info(f"{completed}/{len(files)} files completed.")
     pool.close()
-    pool.join()
```
we can keep this line I think

spenczar replied:
Yes, you're right.

spenczar committed:
Pin to SQLAlchemy<2

SQLAlchemy 2.0 is incompatible with Dagster, which we use
for running precovery jobs. Unfortunately, this means
we need to pin to v1.x.

spenczar committed:
Use v1.x sqlalchemy type definitions with mypy

spenczar committed:
Use sqlalchemy future API for 2.0 compatibility
